src/components/application_manager/test/reset_timeout_handler_test.cc

```diff
@@ -0,0 +1,175 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "application_manager/reset_timeout_handler_impl.h"
+#include "sdl_rpc_plugin/commands/hmi/on_reset_timeout_notification.h"
+#include "sdl_rpc_plugin/commands/mobile/subscribe_way_points_request.h"
+#include "application_manager/mock_application_manager.h"
+#include "application_manager/mock_event_dispatcher.h"
+#include "application_manager/commands/command_request_test.h"
+
+namespace test {
+namespace components {
+namespace reset_timeout_handler_test {
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::NiceMock;
+using application_manager::event_engine::Event;
+using namespace application_manager::commands;
+using namespace application_manager::strings;
+using namespace application_manager::request_controller;
+using sdl_rpc_plugin::commands::hmi::OnResetTimeoutNotification;
+using sdl_rpc_plugin::commands::SubscribeWayPointsRequest;
+
+typedef std::shared_ptr<OnResetTimeoutNotification> NotificationPtr;
+
+const uint32_t kDefaultTimeout = 10000u;
+const uint32_t kTimeout = 3000u;
+const uint32_t kRequestId = 22;
+
+class ResetTimeoutHandlerTest : public commands_test::CommandRequestTest<
+                                    commands_test::CommandsTestMocks::kIsNice> {
+ public:
+  ResetTimeoutHandlerTest() {}
+
+ protected:
+  std::shared_ptr<ResetTimeoutHandlerImpl> reset_timeout_handler_;
+  virtual void SetUp() OVERRIDE {
+    ON_CALL(app_mngr_, event_dispatcher())
+        .WillByDefault(ReturnRef(event_dispatcher_));
+    reset_timeout_handler_ =
+        std::make_shared<ResetTimeoutHandlerImpl>(app_mngr_);
+  }
+};
+
+TEST_F(ResetTimeoutHandlerTest, on_event_OnResetTimeout_success) {
+  std::shared_ptr<SubscribeWayPointsRequest> command =
+      CreateCommand<SubscribeWayPointsRequest>();
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+  ON_CALL(app_mngr_, IsAppSubscribedForWayPoints(_))
+      .WillByDefault(Return(false));
+  ON_CALL(app_mngr_, IsAnyAppSubscribedForWayPoints())
+      .WillByDefault(Return(false));
+  ON_CALL(app_mngr_, GetNextHMICorrelationID())
+      .WillByDefault(Return(kRequestId));
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).WillOnce(Return(true));
+
+  Event event(hmi_apis::FunctionID::BasicCommunication_OnResetTimeout);
+  smart_objects::SmartObject notification_msg;
+  notification_msg[msg_params][reset_period] = kTimeout;
+  notification_msg[msg_params][request_id] = kRequestId;
+  event.set_smart_object(notification_msg);
+
+  ON_CALL(app_mngr_, get_settings())
+      .WillByDefault(ReturnRef(app_mngr_settings_));
+  ON_CALL(app_mngr_settings_, default_timeout())
+      .WillByDefault(ReturnRef(kDefaultTimeout));
+
+  EXPECT_CALL(app_mngr_, GetResetTimeoutHandler())
+      .WillOnce(ReturnRef(*reset_timeout_handler_));
+  EXPECT_CALL(app_mngr_, updateRequestTimeout(_, _, kTimeout));
```

Comment:
@LitvinenkoIra why don't you check first 2 arguments of expectation ?

Reply:
@LuxoftAKutsan fixed 7d67c458fed62cd11e953954fa6285e19842d46c